### PR TITLE
implementing custom icons for thumb element.

### PR DESCRIPTION
**Pull request by snleu**

### CODE REVIEW

File: example/example.py [hunk 1]
```diff
@@ -179,6 +179,8 @@ def set_text_val(val):
     menu.append(ui.Text_Input('mytext',foo,setter=set_text_val))
     gui.append(menu)
 
+    gui.append(ui.Thumb('mytext',foo,label=unichr(0xf04b).encode('utf-8'),hotkey='r',use_icon_font=True,icon_offset_x=5,icon_offset_y=20,icon_scaling_num=1))
+
```
Comment (snleu) [inline example/example.py]: Test 2

Comment (snleu) [inline example/example.py]: Test 3

File: pyglui/ui_elements.pxi [hunk 2]
```diff
@@ -929,25 +938,51 @@ cdef class Thumb(UI_element):
             utils.draw_circle(self.button.center,radius=int(min(self.button.size)*.77), stroke_width = stroke_width, color=RGBA(*thumb_color_shadow),sharpness=thumb_button_shadow_sharpness)
             utils.draw_circle(self.button.center,radius=int(min(self.button.size)*.77), stroke_width = stroke_width, color=self.off_color,sharpness=thumb_button_sharpness)
 
+        print 'button center: ', self.button.center
+        print 'radius: ', int(min(self.button.size)*.77)
+        print 'stroke_width: ', stroke_width
```
Comment (papr) [inline pyglui/ui_elements.pxi]: Use `logger.debug(<string>)` for consistent logging messages. `print` statements will always be displayed. Debug messages are only shown if the logger is configured that way.

File: pyglui/ui_elements.pxi [hunk 3]
```diff
@@ -929,25 +938,51 @@ cdef class Thumb(UI_element):
...
 
         glfont.push_state()
+
+        # Choose either icon (fontawesome) or letter(roboto) label for button
+        cdef bytes text_to_draw
+        cdef float size_of_text
+        cdef list text_center_point
+
         glfont.set_font('roboto')
+        text_to_draw = self.label[0]
```
Comment (papr) [inline pyglui/ui_elements.pxi]: Get rid of `text_to_draw` and always draw `self.label`. If people draw strings longer than one character, it is their fault.

File: pyglui/ui_elements.pxi [hunk 4]
```diff
@@ -929,25 +938,51 @@ cdef class Thumb(UI_element):
...
+
+        if self.use_icon_font:
+            glfont.set_font('fontawesome')
+            text_to_draw = self.label
+            size_of_text = 0.7*max(1,int(min(self.button.size))-thumb_font_padding)
+            text_center_point[0] = self.button.center[0] + self.icon_offset_x
+            text_center_point[1] = self.button.center[1] + self.icon_offset_y
+            if self.icon_scaling_num > 0:
+                size_of_text = self.icon_scaling_num*max(1,int(min(self.button.size))-thumb_font_padding)
+
```
Comment (papr) [inline pyglui/ui_elements.pxi]: Make `icon_scaling_num` default to `1.0`. Then you can get rid of the if-statement and line `966`.

File: pyglui/ui_elements.pxi (hunk 2 above)
Comment (papr) [inline pyglui/ui_elements.pxi]: def9ac4 solves this problem.

Comment (snleu) [inline pyglui/ui_elements.pxi]: Just pushed with the print statements deleted, will use logger.debug in future

**Comment by mkassner** [comment]
closed in favour of alternative PR.

**Pull request closed without merging**